### DISCUSSION TITLE
self-hosting/installation/pterodactyl/

### DISCUSSION THREAD

**Discussion by giscus** [discord-tickets/bot#444] [Documentation]
# self-hosting/installation/pterodactyl/

How to install Discord Tickets with Pterodactyl

https://discordtickets.app/self-hosting/installation/pterodactyl/

**Comment by TheAFKGamer10** [comment]
Is there a way to install on Pterodactyl? There is no instructions in the docs.

**Reply by RooRay** [reply]
Currently, we just recommend using the [nodejs generic](https://github.com/parkervcp/eggs/tree/master/generic/nodejs) egg and uploading the release files to your server and then following the standard configuration. Me and Eartharoid are actively working on making a proper egg. 🙂

**Reply by TheAFKGamer10** [reply]
Ok, Thanks. Can you let me know when you have the egg ready? I will use the standard install for now.

**Reply by RooRay** [reply]
> Currently, we just recommend using the [nodejs generic](https://github.com/parkervcp/eggs/tree/master/generic/nodejs) egg and uploading the release files to your server and then following the standard configuration. Me and Eartharoid are actively working on making a proper egg. 🙂

We're also working on making proper docs for how to use the egg

**Reply by TheAFKGamer10** [reply]
Sounds good.

**Reply by RooRay** [reply]
> Ok, Thanks. Can you let me know when you have the egg ready? I will use the standard install for now.

I'll mention it here but there will also probably be an announcement in the support server and some other places

**Reply by eartharoid** [reply]
I decided against a custom egg but the documentation for using a generic Node.js egg is now available at https://discordtickets.app/self-hosting/installation/pterodactyl/. I finished it a week ago but forgot about this.

**Comment by RjManhas** [comment]
why did you guys chose against a custom egg? @eartharoid

**Reply by eartharoid** [reply]
I wanted to use custom Docker images so that the image dropdown would be used to select the version, but
1. If it included patch versions you would need to update the egg every time (although `4.0`, `latest`, and `main` would be ok)
2. I changed the existing Docker image to make it compatible with Pterodactyl (`container` user, moved the files to `/home/container`), but all of the files get deleted because that directory is a volume. So I'd need to put the code somewhere else (and I don't want to now because I like where it is) and then use the startup script to either copy the files back to `/home/container` or make symlinks for certain files. 

Other than the image/version selector (with `4.0` etc meaning you get patches automatically just by restarting), the main benefit would have been easier configuration using input fields on the Startup page. But for the port and IP address to be filled automatically, those would still need to be in the `.env` file and having some variables in different places seemed dumb.

**Comment by ISaiDPower** [comment]
[ERROR] (HTTP) SvelteKit 1894c2f973d TypeError: fetch failed

when trying to access settings panel

**Reply by eartharoid** [reply]
@ISaiDPower The bot makes requests to itself and this error means that your `HTTP_EXTERNAL` value is not reachable.
Please join https://lnk.earth/discord so we can find out why.

**Comment by Mantelbart** [comment]
Any1 wanna set that up for me?

**Reply by TheAFKGamer10** [reply]
What do you need help with?
